src/pages/User/components/PullRequests/PullRequest/index.js

```diff
@@ -1,20 +1,61 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import MergeStatus from './MergeStatus';
 import PullRequestInfo from './PullRequestInfo';
+import { GITHUB_TOKEN } from '../../../../../config';
+import { fetchInfoFromGitHub } from '../../../../../utils/utils';
 
 const ISSUE_STATUS = {
   ALL: 'all',
-  OPEN: 'open',
-  CLOSED: 'closed'
+  OPEN: 'open'
 };
 
-const PullRequest = ({ pullRequest }) => (
-  <div className={`bg-white leading-normal p-4 flex border-b border-grey break-words`}>
-    <MergeStatus open={pullRequest.state === ISSUE_STATUS.OPEN} merged={pullRequest.state === ISSUE_STATUS.CLOSED} />
-    <PullRequestInfo pullRequest={pullRequest} />
-  </div>
-);
+/**
+ * Pull Request Component.
+ */
+class PullRequest extends Component {
+  state = {
+    isMerged: false
+  };
+
+  /**
+   * Life cycle event for componentDidMount.
+   */
+  componentDidMount = async () => {
+    const { pullRequest } = this.props;
+    const pullNumber = pullRequest.number;
+    const splittedPRUrlArray = pullRequest.html_url.split('/');
+    const owner = splittedPRUrlArray[3];
+    const repo = splittedPRUrlArray[4];
+    const apiUrl = `https://api.github.com/repos/${owner}/${repo}/pulls/${pullNumber}/merge`;
+
+    try {
+      const merged = await fetchInfoFromGitHub(apiUrl, GITHUB_TOKEN);
+
+      this.setState({
+        isMerged: merged.ok
+      });
+    } catch (e) {
+      this.setState({
+        isMerged: false
```

Comment:
I’m doubtful here. Exception could be anything and this false can result into ‘closed’ PR state right? Most of the case it will work. Maybe adding tests would make it more valid 🙂